todos/src/main/java/com/maksimpegov/todos/TodosService.java

```diff
@@ -1,35 +1,95 @@
 package com.maksimpegov.todos;
 
+import com.maksimpegov.todos.models.ProcessResponse;
 import com.maksimpegov.todos.models.Todo;
 import com.maksimpegov.todos.models.TodoRepository;
 import org.springframework.stereotype.Service;
 
-import java.lang.reflect.Array;
+import java.util.Collections;
+import java.util.Date;
 import java.util.List;
 
 @Service
 public record TodosService(TodoRepository todoRepository) {
-    public List<Todo> getTodos(String userId) {
-//        List<Todo> todos = todoRepository.getAllByUserId(userId);
-//        return todos;
-        List<Todo> todos = List.of(new Todo("Sleep", "1", new java.util.Date()), new Todo("Eat", "2", new java.util.Date()));
-        return todos;
+    public ProcessResponse getTodos(String userId) {
+        ProcessResponse response = new ProcessResponse();
+        try {
+            List<Todo> todos = todoRepository.getAllByUserId(userId);
+            response.setTodos(todos);
+            response.setStatus("200");
+        } catch (Exception e) {
+            response.setStatus("404");
+        }
+        return response;
     }
 
-    public String addTodo(String userId, Todo todo) {
-        Todo todo1 = new Todo("Sleep", "1", new java.util.Date());
-        todoRepository.save(todo1);
-        return "OK";
+    public ProcessResponse addTodo(Todo todo) {
+        ProcessResponse response = new ProcessResponse();
+        try {
+            todo.setCreatedAt(new Date());
+            todoRepository.save(todo);
+            response.setTodos(Collections.singletonList(todo));
+            response.setStatus("200");
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            response.setStatus("400");
+        }
+        return response;
     }
 
-    public String deleteTodo(String userId, String todoId) {
-        todoRepository.deleteById(Long.parseLong(todoId));
-        return "OK";
+    public ProcessResponse editTodo(Todo newTodo) {
+        ProcessResponse response = new ProcessResponse();
+        try {
+            Todo oldTodo = todoRepository.getOne(newTodo.getId());
+            oldTodo.setCreatedAt(new Date());
+            if (newTodo.getText() != null) {
+                oldTodo.setText(newTodo.getText());
+            } else if (newTodo.getClosedAt() != null) {
```

Comment:
this whole part with if else if is looks strange... It is needed only to merge two objects of class ToDo, rignt?
If yes, then the code is really looks bad. There is no way to understend from the code that newTodo is only partial ToDo. You need another way to merge newTodo and oldTodo objects.